components/eamxx/src/control/atmosphere_driver.cpp

```diff
@@ -870,22 +870,26 @@ void AtmosphereDriver::set_initial_conditions ()
       }
     } else if (not (fvphyshack and grid_name == "Physics PG2")) {
       auto& this_grid_ic_fnames = ic_fields_names[grid_name];
-      auto& this_grid_topo_fnames_nc = topography_fields_names_nc[grid_name];
-      auto& this_grid_topo_fnames_eamxx = topography_fields_names_eamxx[grid_name];
+      auto& this_grid_topo_file_fnames = topography_file_fields_names[grid_name];
+      auto& this_grid_topo_eamxx_fnames = topography_eamxx_fields_names[grid_name];
 
       auto c = f.get_header().get_children();
 
       if (fname == "phis") {
         // Topography (phis) is a special case that should
-        // be loaded from the topography file (assuming
-        // input files doesn't contain a value).
-        if (fvphyshack) {
-          this_grid_topo_fnames_nc.push_back("PHIS_d");
-          this_grid_topo_fnames_eamxx.push_back("phis");
+        // be loaded from the topography file, where the
+        // eamxx field "phis" corresponds to the name
+        // "PHIS_d" on the GLL and Point grids and "PHIS"
+        // on the PG2 grid in the topography file.
+        if (grid_name == "Physics PG2") {
+          this_grid_topo_file_fnames.push_back("PHIS");
+        } else if (grid_name == "Physics GLL" ||
+                   grid_name == "Point Grid") {
```

Comment:
Side comment: `Point Grid` is just for unit tests, right? At some point we should make this more explicit, so we clearly recognize pieces of code that are not relevant for normal runs...

Reply:
Yeah, I was going to have a separate `else if (PointGrid)`, but it duplicated code. I'm willing to put it back in.

Reply:
No, it's fine. I just wanted to make sure we were not using this kind of generic/confusing names in production code.